src/clients/client.ts

```diff
@@ -120,3 +123,97 @@ export function useClientCallback(clientType?: ClientType): Client {
   }, [clientFromRef]);
   return clientFromRef;
 }
+
+export type FetchStatus =
+  | 'unauthorized'
+  | 'ready'
+  | 'loading'
+  | 'error'
+  | 'loaded'
+  | 'waiting';
+
+export type ApiAccessTokenActions = {
+  fetch: (options: FetchApiTokenOptions) => Promise<JWTPayload | FetchError>;
+  getStatus: () => FetchStatus;
+  getTokens: () => JWTPayload | undefined;
+};
+
+export const ApiAccessTokenActionsContext = createContext<ApiAccessTokenActions | null>(
+  null
+);
+
+export function useApiAccessTokens(): ApiAccessTokenActions {
+  const client = useClient();
+  const tokens = client.isAuthenticated() ? client.getApiTokens() : undefined;
+  const hasTokens = tokens && Object.keys(tokens).length;
+  const [apiTokens, setApiTokens] = useState<JWTPayload | undefined>(
+    hasTokens ? tokens : undefined
+  );
+
```

Comment:
Resolving current status of this hook, is complicated because oidc-client status may change at any time. Also the internal status may change any time with user actions.